Models/EventTypesListModel.h

```diff
@@ -0,0 +1,30 @@
+#ifndef EVENTTYPESSLISTSMODEL_H
+#define EVENTTYPESSLISTSMODEL_H
+
+#include "EventDescriptor.pb.h"
+#include "RepeatedMessageModel.h"
+#include "event_reader/event_parser.h"
+
+#include <QAbstractListModel>
+#include <QIcon>
+#include <QSortFilterProxyModel>
+
+class EventTypesListModel : public QAbstractListModel {
+ private:
+  EventData* eventData_;
+
+ public:
+  static enum UserRoles {
+    EventTypeRole = Qt::UserRole,
+    EventGroupRole = Qt::UserRole + 1,
+    EventArgumentsRole = Qt::UserRole + 2,
+    EventBareIDRole = Qt::UserRole + 3
```

Comment:
I don't think you need to do this, I think in C++ enums you only have to specify the starting index.

>An enumerator-definition without an initializer gives the enumerator the value obtained by increasing the value of the previous enumerator by one.
https://stackoverflow.com/a/34811571

```suggestion
    EventGroupRole,
    EventArgumentsRole,
    EventBareIDRole
```